src/core/oauth2-authorize.js

```diff
@@ -78,3 +66,30 @@ export default function authorize ( { auth, authActions, errActions, configs, au
 
   win.open(url)
 }
+
+function processUrl(url, authConfigs, query=[]) {
+  let result = url || ""
+  if (authConfigs) {
+    if (authConfigs.realm) {
+      const placeholder = "{realm}"
+      const idx = url ? url.indexOf(placeholder) : -1
+      if (idx !== -1) {
+        result = url.substring(0, idx) + encodeURIComponent(authConfigs.realm) + url.substring(idx + placeholder.length)
+      } else {
+        query.push("realm=" + encodeURIComponent(authConfigs.realm))
+      }
+    }
+    const params = authConfigs.additionalQueryStringParams || {}
+    for (let key in params) {
+      if (params[key] !== undefined) {
+        query.push([key, params[key]].map(encodeURIComponent).join("="))
+      }
+    }
+    if (query.length) {
+      result += (result.indexOf("?") === -1 ? "?" : "&") + query.join("&")
+    }
+  }
+  return result
+}
```

Comment:
This is the meat of it. The existing code to fill in the configs and additional query params is now in this function, where I also added the bit that does the replacement of the placeholder.